lsp/nls/src/requests/hover.rs

```diff
@@ -156,57 +160,47 @@ pub fn handle(
     if let Some(hover) = hover_data {
         let mut contents = Vec::new();
 
-        // If we can't determine a static type through the typechecker because we are outside of a
-        // statically typed block, but the term points to a definition with a type annotation, we
-        // use this annotation insted.
-        let mut type_annots: Vec<_> = hover
+        // Collect all the type and contract annotations we can find. (We don't distinguish between them.)
+        let mut annotations: Vec<_> = hover
             .metadata
             .iter()
-            .filter_map(|m| Some(m.annotation.typ.as_ref()?.typ.to_string()))
+            .flat_map(|m| {
+                let maybe_typ = m.annotation.typ.as_ref().map(|typ| typ.typ.to_string());
+                let contracts = m
+                    .annotation
+                    .contracts
+                    .iter()
+                    .map(|c| c.label.typ.to_string());
+                maybe_typ.into_iter().chain(contracts)
```

Comment:
You have an `iter()` function on `TypeAnnotation` that chains the type and then the contracts, so that you don't have to do it manually here (https://github.com/tweag/nickel/blob/3bafeff167b9130b43567f235a627dbad9d29d76/core/src/term/mod.rs#L758)